crates/iota-json-rpc/src/error.rs

```diff
@@ -192,12 +184,44 @@ impl From<Error> for RpcError {
                     }
                     QuorumDriverError::ObjectsDoubleUsed {
                         conflicting_txes,
-                        retried_tx,
-                        retried_tx_success,
+                        retried_tx_status,
                     } => {
+                        let weights: Vec<u64> =
+                            conflicting_txes.values().map(|(_, stake)| *stake).collect();
+                        let remaining: u64 = TOTAL_VOTING_POWER - weights.iter().sum::<u64>();
+
+                        // better version of above
+                        let reason = if weights.iter().all(|w| remaining + w < QUORUM_THRESHOLD) {
+                            "equivocated until the next epoch"
+                        } else {
+                            "reserved for another transaction"
+                        };
+
+                        let retried_info = match retried_tx_status {
+                            Some((digest, success)) => {
+                                format!(
+                                    "Retried transaction {} ({}) because it was able to gather the necessary votes.",
+                                    digest,
+                                    if success { "succeeded" } else { "failed" }
+                                )
+                            }
+                            None => "".to_string(),
+                        };
+
                         let error_message = format!(
-                            "Failed to sign transaction by a quorum of validators because of locked objects. Retried a conflicting transaction {:?}, success: {:?}",
-                            retried_tx, retried_tx_success
+                            "Failed to sign transaction by a quorum of validators because one or more of its objects is {}. {} Other transactions locking these objects:\n{}",
+                            reason,
+                            retried_info,
+                            conflicting_txes
+                                .iter()
+                                .sorted_by(|(_, (_, a)), (_, (_, b))| b.cmp(a))
+                                .map(|(digest, (_, stake))| format!(
+                                    "- {} (stake {}.{})",
+                                    digest,
+                                    stake / 100,
+                                    stake % 100,
+                                ))
+                                .join("\n"),
```

Comment:
Nit: This could be wrapped into a `QuoromDriveError` method. E.g. `to_error_message`.

Reply:
I added a commit to move that logic.